core/src/main/java/feign/Logger.java

```diff
@@ -89,7 +90,9 @@ protected Response logAndRebufferResponse(String configKey, Level logLevel, Resp
       }
 
       int bodyLength = 0;
-      if (response.body() != null) {
+      if (response.body() != null && !(status == 204 || status == 205)) {
```

Comment:
I'm using OkHttpClient. There was similar issue in retrofit and from what I can tell they decided that this is not okhttp bug: https://github.com/square/retrofit/issues/735